lib/cli-engine/config-array-factory.js

```diff
@@ -697,6 +699,8 @@ class ConfigArrayFactory {
             ctx.matchBasePath
         );
 
+        const clonedRulesConfig = rules && JSON.parse(JSON.stringify((rules)));
```

Comment:
Would you add a comment with the reason we clone it?

Reply:
Can you check if the description [here](https://github.com/eslint/eslint/pull/13034/files#diff-91937c0f974645b62296ca160b952f7eR700-R724) is fine or not. ?

Reply:
Awesome. It's very detailed!